runtime/autoload/man.vim

```diff
@@ -118,15 +118,12 @@ endfunction
 " Expects a string like 'access' or 'access(2)'.
 function s:parse_page_and_section(sect, str) abort
   try
-    let save_isk = &iskeyword
-    setlocal iskeyword-=(,)
-    let page = substitute(a:str, '(*\(\k\+\).*', '\1', '')
-    let sect = substitute(a:str, '\(\k\+\)(\([^()]*\)).*', '\2', '')
+    let page = substitute(a:str, '\([-.[:alnum:]_]\+\).*', '\1', '')
+    let sect = substitute(a:str, '\([-.[:alnum:]_]\+\)(\([^()]*\)).*', '\2', '')
     if sect == page || -1 == match(sect, '^[0-9 ]\+$')
```

Comment:
> By the way, what with sections like {number}p and {letter}?

I wasn't aware of this, but yes, this should be accommodated. I'm only able to find examples in the forms

- `{number}`
- `{number}{letter}`
- `{letter}`

but I suppose there is no point in restricting the pattern to these options.

> This makes test if 0 + a:sect > 0 invalid.

The section name is assumed to be numeric in several places throughout the script, including `man#get_page()`. Changing this makes the single-argument version of the function ambiguous: the argument could either be a page name or section name (in which case the page is set to `expand('cword')`).

The function is referenced in `ftplugin/man.vim` with one argument:

```vim
nnoremap <silent> <buffer> <C-]> :call man#get_page(v:count)<CR>
" ...
nnoremap <silent> <buffer> K :call man#get_page(v:count)<CR>
```
where `[count]` can be used to specify the section number. I propose the following change:

```vim
nnoremap <silent> <buffer> <C-]> :call man#get_page(expand('<cword>'), v:count)<CR>
" ...
nnoremap <silent> <buffer> K :call man#get_page(expand('<cword>'), v:count)<CR>
```

And to always interpret a single argument as page name. I think this is easier to understand anyway ("it takes one required argument, the page name, and an optional second argument, the section name"). I realize `v:count` *must* be numeric, alleviating the ambiguity somewhat, but it shouldn't be assumed that `man#get_page()` will only be used in this context.